### PR TITLE
Create a test suite package

### LINKED ISSUE #67: Add a testsuite
At the moment this library has pretty much no tests. There are a few examples (`oneoftest`,`addressbook`) That could be a good start one.

 @sjking you volunteered to help out with this?

### PR DESCRIPTION
**Pull request by sjking**
This test suite makes use of the Makefile. To run use: `make test`. Resolves #67.

### CODE REVIEW

File: protobuf-test-suite/codegen.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+display_modules () {
+  for i in ${1}; do
+    printf "\n%"$2"s$i";
+  done
+}
+
+CABAL_FILE="protobuf-test-suite.cabal"
+
+OTHER_MODULES=""
+[ -f "$CABAL_FILE" ] && OTHER_MODULES=`egrep 'other-modules' "$CABAL_FILE" -A10000`
```
Comment (tvh) [inline protobuf-test-suite/codegen.sh]: Do we need to generate code like this? I think hpack https://github.com/sol/hpack might be a nicer way to do this.

File: stack-test.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+packages:
+- '.'
+- descriptor/
+- hprotoc/
+- protobuf-test-suite/
+extra-deps: []
+resolver: nightly-2018-05-21
```
Comment (tvh) [inline stack-test.yaml]: I think we should not have/need an extra stack.yaml for this. That way we could run the testsuite with travis on multiple GHC versions.

**Comment by tvh** [comment]
Nice work so far. Can you tell when you are happy with this? As far as I can tell there are a few things that would be nice, but could also be adressed in a different PR.
- [ ] `oneof` There is some code in `hprotoc/oneoftest`
- [ ] recursuve types/hs-boot
- [ ] All builtin types. There is Code here in `tests/Arb/`

**Comment by sjking** [comment]
I found a problem in how the Haskell code was generated for `oneof` but found a solution with sjking@3f0d701. It might be a good time to merge this fix. I will still look into adding tests for recursive types/hs-boot and look into testing the built-in types after that.

**Comment by tvh** [comment]
Cool, I will merge it now then.

**Pull request merged**